modules/nixvim/nixvim.nix

```diff
@@ -63,80 +83,78 @@ in
     })
   ];
 
-  config =
-    lib.mkIf (config.stylix.enable && cfg.enable && (config.programs ? nixvim))
-      (
-        lib.optionalAttrs (builtins.hasAttr "nixvim" options.programs) (
-          lib.mkMerge [
-            (lib.mkIf (cfg.plugin == "base16-nvim") {
-              programs.nixvim.colorschemes.base16 = {
-                enable = true;
+  config = lib.mkIf (config.stylix.enable && cfg.enable) {
+    stylix.targets.nixvim.config = lib.mkMerge [
```

Comment:
This option def shouldn't need to be conditional.

Rather, only the `programs` definition below should be conditional.

For comparison, [my impl](https://github.com/danth/stylix/pull/415/files#diff-007d386b8b3b21d785f2b5c6b3d437fcc8785c0b9ebc833dd4bd49ee2f9a8e84R66-R72) looks like:

```nix
config = {
  programs = lib.mkIf (config.stylix.enable && cfg.enable) (
    lib.optionalAttrs (options.programs ? nixvim) {
      nixvim = config.lib.stylix.nixvim.config;
    }
  );
  lib.stylix.nixvim.config = lib.mkMerge [
```